iroh/src/baomap/flat.rs

```diff
@@ -799,7 +938,94 @@ impl Store {
         if size < self.0.options.inline_threshold {
             state.data.insert(hash, data.to_vec().into());
         }
-        Ok(hash)
+        drop(complete_io_guard);
+        Ok(cid)
+    }
+
+    fn delete_sync(&self, hash: Hash) -> io::Result<()> {
+        let mut data = None;
+        let mut outboard = None;
+        let mut external = None;
+        let mut partial_data = None;
+        let mut partial_outboard = None;
+        let complete_io_guard = self.0.complete_io_mutex.lock().unwrap();
+        let mut state = self.0.state.write().unwrap();
+        if let Some(entry) = state.complete.remove(&hash) {
+            if entry.owned_data {
+                data = Some(self.owned_data_path(&hash));
+            }
+            if needs_outboard(entry.size) {
+                outboard = Some(self.owned_outboard_path(&hash));
+            }
+            if !entry.external.is_empty() {
+                external = Some(self.0.options.paths_path(hash));
+            }
+        }
+        if let Some(partial) = state.partial.remove(&hash) {
+            partial_data = Some(self.0.options.partial_data_path(hash, &partial.uuid));
+            if needs_outboard(partial.size) {
+                partial_outboard = Some(self.0.options.partial_outboard_path(hash, &partial.uuid));
+            }
+        }
+        state.outboard.remove(&hash);
+        state.data.remove(&hash);
+        drop(state);
+        if let Some(data) = data {
+            if let Err(cause) = std::fs::remove_file(data) {
+                tracing::warn!("failed to delete data file: {}", cause);
+            }
+        }
+        if let Some(external) = external {
```

Comment:
does this delete external files?  or is this just the outboard?

Reply:
There is a file that contains the paths for the external files. It deletes that file, not the external file itself. Deleting external files would be quite evil.